src/OrchardCore/OrchardCore.BackgroundTasks.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,18 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace OrchardCore.BackgroundTasks
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddBackgroundTaskAttributes(this IServiceCollection services)
+        {
+            services.TryAddEnumerable(
```

Comment:
nit: if you don't need DI, having a `BackgroundTaskAttributeOptionsSetup` is probably not necessary. You can do something similar inline:

```csharp
services.Configure<BackgroundTaskOptions>(options =>
{
    options.SettingsProviders.Add(new BackgroundTaskAttributeSettingsProvider());
});
```